flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java

```diff
@@ -367,6 +368,46 @@
                                             code("FlinkKubeClient#checkAndUpdateConfigMap"))
                                     .build());
 
+    public static final ConfigOption<String> JOB_MANAGER_POD_TEMPLATE =
+            key("kubernetes.jobmanager.pod-template-file")
+                    .stringType()
+                    .noDefaultValue()
+                    .withFallbackKeys(KUBERNETES_POD_TEMPLATE_FILE_KEY)
+                    .withDescription(
+                            "Specify a local file that contains the jobmanager pod template. It will be used to "
+                                    + "initialize the jobmanager pod. If not explicitly configured, config option '"
+                                    + KUBERNETES_POD_TEMPLATE_FILE_KEY
+                                    + "' will be used.");
+
+    public static final ConfigOption<String> TASK_MANAGER_POD_TEMPLATE =
+            key("kubernetes.taskmanager.pod-template-file")
+                    .stringType()
+                    .noDefaultValue()
+                    .withFallbackKeys(KUBERNETES_POD_TEMPLATE_FILE_KEY)
+                    .withDescription(
+                            "Specify a local file that contains the taskmanager pod template. It will be used to "
+                                    + "initialize the taskmanager pod. If not explicitly configured, config option '"
+                                    + KUBERNETES_POD_TEMPLATE_FILE_KEY
+                                    + "' will be used.");
+
+    /**
+     * This option is here only for documentation generation, it is the fallback key of
+     * JOB_MANAGER_POD_TEMPLATE and TASK_MANAGER_POD_TEMPLATE.
+     */
+    @SuppressWarnings("unused")
+    public static final ConfigOption<String> KUBERNETES_POD_TEMPLATE =
+            key(KUBERNETES_POD_TEMPLATE_FILE_KEY)
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
```

Comment:
Shall we explain in the configuration option description that the template should use `Constants.MAIN_CONTAINER_NAME` to refer the main container?